django/db/models/fields/__init__.py

```diff
@@ -1286,14 +1286,14 @@ def get_internal_type(self):
     def to_python(self, value):
         if value is None:
             return value
-        if isinstance(value, datetime.datetime):
+        if hasattr(value, 'tzinfo') and hasattr(value, 'date'):
```

Comment:
Perhaps this would be more readable with a preceding comment such as
`# if isinstance(value, datetime.datetime) but supports mock datetimes too.`